site/content/docs/browser-checks/github-code-sync.md

```diff
@@ -0,0 +1,119 @@
+
+
+## Linking file(s) from a repo
+
+After installing the Checkly GitHub App, the first step is to pick a repository and link an entry file to a Browser check. 
+Create a new Browser check or navigate to an existing one and click the "Sync from GitHub" button.
+
+In the GitHub integration section...
+
+1. Toggle **Sync from GitHub**.
+2. Pick a repository with the file you want to sync.
+3. You will now choose your entry file and syncing strategy.
+
+> Note that your **entry file** can `require` other local file dependencies. Checkly will fetch them recursively.
+See [Working with local dependencies](#working-with-local-dependencies) below for more details.
+
+## Sync on commit to a branch
+
+The default syncing strategy is to sync your code when committing to a specific branch you select, i.e. `main`. This can 
+be a direct `git push` to `main` or when merging a Pull Request from a development branch into `main`. To use this strategy:
+
+1. Pick the target branch. Commits to this branch will trigger the sync.
+2. Type out the path to the entry file you want to sync.
+
+__insert image here__
+
+What are the pro's and cons of this strategy?
+
+- **Pros:** Simplicity. Just push or merge and Checkly syncs and persists your code.
+- **Cons:** A commit to `main` normally does not coincide with a deployment being done and live. This means you could update
+your check code, before or potentially after any changes you made to your app are live: the check code is out of sync for a 
+period with your application.
+
+## Sync on deployment
+
+The "Sync on Deployment" strategy allows you to sync your checks' code when Checkly receives a "deployment done" event from 
+one of the deployment providers we support, currently [Vercel](/docs/cicd/vercel/), [GitHub Deployments](/docs/cicd/github/) 
+and [Command Line Triggers](/docs/cicd/triggers/).
+
+But it's a bit smarter than just that. With this strategy you can:
+
+1. Trigger checks against Preview or staging environments, using your checks' code from any branch.
+2. Sync your checks' code *only* when your Production deployment passes, so you never have outdated check running against a fresh deployment.
+
+To use this strategy:
+
+1. Pick the **target environment name**, e.g. `Production`. This name determines when Checkly should sync and persist your code, 
+in contrast to syncing ephemerally. More on that below.
+2. Type out the path to the entry file you want to sync, just like the "Sync on Commit" strategy.
+
+Now you need to make sure you have integrated one of our supported integrations that can tell Checkly a deployment is done. 
+Let's use the [Vercel](/docs/cicd/vercel/) integration as an example.
+
+After setting up the Vercel integration, two workflows open up.
+
+**1. Working on Preview branches**
+
+You push an updated app to a development branch. Vercel proceeds to deploy a `Preview` environment and Checkly triggers
```

Comment:
This is actually called `staging` on the vercel side

Reply:
@umutuzgur hmmm, I don't think a user is exposed to that information. They only see `Preview` and `Production` 
![CleanShot 2022-08-04 at 11 19 41@2x](https://user-images.githubusercontent.com/3802923/182811814-3004655e-4f13-49ba-bfc8-3c8fde326e88.png)

Reply:
This is a reason why I thought about having a sync per integration. So we can make it look nicer and get rid of some complexity for customers